Source/controls/plrctrls.cpp

```diff
@@ -534,7 +534,7 @@ void Interact()
 	}
 
 	if (leveltype != DTYPE_TOWN && PlayerUnderCursor != nullptr && !myPlayer.friendlyMode) {
-		NetSendCmdParam1(true, myPlayer.UsesRangedWeapon() ? CMD_RATTACKPID : CMD_ATTACKPID, PlayerUnderCursor->getId());
+		NetSendCmdParam1(true, myPlayer.UsesRangedWeapon() ? CMD_RATTACKPID : CMD_ATTACKPID, static_cast<uint16_t>(PlayerUnderCursor->getId()));

```

Comment:
> We had some discussion about the type of the id's in #6791.
> 
> We could change `getId()` to `uint8_t` but should we also update all code where we currently use `size_t` for player ids to `uint8_t`?

integer promotion doesn't trigger a warning so it's less of a priority to do that sort of change, it'd be a nice to have (might even make an impact on code size for constrained platforms?)

> Thankfully we changed many functions to Player references, so there aren't that many left. Btw: when changing to `uint8_t` warnings get down by 10 (from 101 to 91).
> 
> For now I'm not sure if we should simply merge this as it is and refactor this in a secondary step there aren't that many cases where player id casting happens.

I mainly brought it up here as in the current master `Player::getId()` is explicitly cast 4 times, this PR introduces another 9 casts (+2 for `MyPlayerId`). You could instead remove 3 of the casts and only leave the conversion to int16_t in `Player::occupyTile` as every other use should promote without warning.

Reply:
We have some functions that don't use `getId()` bust instead have a `size_t playerId` parameter and need to be casted (for example `NetSendHiPri`).

I ask more how much consistence do we want to have in the codebase
1. Change `getId()` to `unit8_t`
2. Change `getId()` and `MyPlayerId` to `unit8_t`
3. Change `getId()`, `MyPlayerId` and all function parameters (`playerId`) to `unit8_t`
4. Leave everything `size_t`

Mostly I try to avoid this question with passing Player reference around instead of using the ids, but in the network code this is not possible. 😢 

The question is: how should we proceed now? I have no strong opinion on it. I want to get rid of the warnings one way or another. 😁

Reply:
Somewhere between 2-3 for now I think :)

1 is also acceptable for a first PR if that is less complicated

Reply:
I picked 2 for now.
Hopefully we can change more functions to `Player` reference.